packages/core/strapi/lib/types/utils/guard.d.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Assign a default value `TDefault` to `TValue` if `TValue` is of type `never`
+ *
+ * @example
+ * type X = Never<{ foo: 'bar' }, string>
+ * // { foo: 'bar' }
+ *
+ * type X = Never<never>
+ * // unknown
+ *
+ * type X = Never<never, string>
+ * // string
+ */
+export type Never<TValue, TDefault = unknown> = [TValue] extends [never] ? TDefault : TValue;
```

Comment:
I find `Never` a confusing name, it's more like `Coalesce` or fallback/default

Reply:
That would be the name of the module then, no? :thinking: 
Here we're defining a guard that fallback to a default if the given type is never. In theory, we could add other guards for other types (like null or other)

Reply:
Ok, I guess it's clear enough when used in the context of Utils.Guard.Never, I'm fine with it like this